contracts/masset/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,263 @@
+
+        emit LiquidationAdded(_bAsset);
+    }
+
+    /**
+     * @dev Get a liquidation
+     * @param   _bAsset Address for the underlying bAsset
+     * @return  liquidation The liquidation data
+     */
+    function readLiquidation(address _bAsset)
+        external
+        onlyGovernor
+        returns (Liquidation memory liquidation)
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+
+        liquidation = liquidations[_bAsset];
+        return liquidation;
+    }
+
+    /**
+    * @dev Update a liquidation
+    * @param _bAsset The _bAsset address that this liquidation is for
+    * @param _integration The integration contract address for the _bAsset
+    * @param _sellToken The token address to be sold
+    * @param _uniswapPath The Uniswap path as an array of addresses e.g. [COMP, WETH, DAI]
+    * @param _amount The amount of tokens to be sold
+    * @param _paused Whether the liquidation is paused
+    */
+    function updateLiquidation(
+        address     _bAsset,
+        address     _integration,
+        address     _sellToken,
+        address[]   calldata _uniswapPath,
+        uint        _amount,
+        bool        _paused
+    )
+
+        external
+        onlyGovernor
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+
+        Liquidation storage liq = liquidations[_bAsset];
+        liq = liquidations[_bAsset];
+        liq.bAsset = _bAsset;
+        liq.integration = _integration;
+        liq.sellToken = _sellToken;
+        liq.uniswapPath = _uniswapPath;
+        liq.amount = _amount;
+        liq.paused = _paused;
+
+        emit LiquidationPaused(_bAsset);
+    }
+
+    /**
+    * @dev Delete a liquidation
+    * @param _bAsset The _bAsset for the liquidation
+    */
+    function deleteLiquidation(address _bAsset)
+        external
+        onlyGovernor
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+
+        delete liquidations[_bAsset];
+        emit LiquidationRemoved(_bAsset);
+    }
+
+    /**
+    * @dev  Update the Uniswap contract address
+    *       Whilst it is unlikely this will be needed it is helpful for testing
+    * @param _uniswapAddress The uniswap address to assign
+    */
+    function updateUniswapAddress(address _uniswapAddress)
+        external
+        onlyGovernor
+    {
+
+        require(_uniswapAddress != address(0), "_uniswapAddress cannot be zero address");
+        uniswapAddress = _uniswapAddress;
+        emit UniswapUpdated(_uniswapAddress);
+    }
+
+    /**
+    * @dev Allows a calling integration contract to collect redeemed tokens
+    */
+    function collect(address _bAsset)
+        external   
+    {
+        Liquidation memory liq = liquidations[_bAsset];
+        require(msg.sender == liq.integration, "Only integration contract can execute");
+
+        address sendAddress = liq.integration;
```

Comment:
We can just use `msg.sender` instead of reading the `integration` here to save gas. OR read `liq.integration` once and save it to a local var, instead of reading twice.